### PR TITLE
Feature/fix mypy errors

### LINKED ISSUE #3537: gf.port.map_ports_layer_to_orientation fails
**Describe the bug**
```bash
731, in map_ports_layer_to_orientation
    layers = {port.layer for port in ports}
              ^^^^^^^^^^
AttributeError: 'str' object has no attribute 'layer'. Did you mean: 'lower'?
```

**To Reproduce**
```python
c = gf.Component()
nxn = gf.components.nxn(west=2, north=2, east=2, south=2)
ref = c << nxn
ports = {p.name: p for p in ref.ports if p.name is not None}
pprint(gf.port.map_ports_layer_to_orientation(ports))
c.show()
```

**Expected behavior**
Not exactly sure, but i think a reasonable output should look like:
```bash
{'1_0_E0': 'o6',
 '1_0_E1': 'o5',
 '1_0_N0': 'o3',
 '1_0_N1': 'o4',
 '1_0_S0': 'o8',
 '1_0_S1': 'o7',
 '1_0_W0': 'o1',
 '1_0_W1': 'o2'}
```

**Suggested fix**
```python

def map_ports_layer_to_orientation(
    ports: "typings.PortDict",
    function: Callable[..., None] = _rename_ports_facing_side,
    **kwargs: Any,
) -> dict[str, str]:
    """Returns dict of port name to port name original.

    Args:
        ports: dict of ports.
        function: to rename ports.
        kwargs: for the function to rename ports.

    .. code::

             N0  N1
             |___|_
        W1 -|      |- E1
            |      |
        W0 -|______|- E0
             |   |
            S0   S1

    """
    m: dict[str, str] = {}
    selected_ports = select_ports(list(ports.values()), **kwargs)
    layers = {port.layer for port in selected_ports}

    for layer in layers:
        direction_ports: PortsDict = {x: [] for x in ["E", "N", "W", "S"]}
        ports_on_layer = [p.copy() for p in selected_ports if p.layer == layer]

        for p in ports_on_layer:
            p.name_original = p.name  # type: ignore
            angle = p.orientation % 360
            if angle <= 45 or angle >= 315:
                direction_ports["E"].append(p)
            elif angle <= 135 and angle >= 45:
                direction_ports["N"].append(p)
            elif angle <= 225 and angle >= 135:
                direction_ports["W"].append(p)
            else:
                direction_ports["S"].append(p)
        layer_tuple = layer if isinstance(layer, kf.LayerEnum) else (layer, 0)
        function(direction_ports, prefix=f"{layer_tuple[0]}_{layer_tuple[1]}_")
        m |= {p.name: p.name_original for p in ports_on_layer}  # type: ignore
    return m
```

### PR DESCRIPTION
**Pull request by MatthewMckee4**
fixes #3537

## Summary by Sourcery

Update type annotations across the codebase.

Bug Fixes:
- Fix mypy errors.

Enhancements:
- Improve type annotations for several functions, including `sort_ports_counter_clockwise`, `select_ports`, `map_ports_layer_to_orientation`, `map_ports_to_orientation_cw`, `auto_rename_ports_layer_orientation`, `clean_value_json`, `get_netlist_recursive`, and `mzi_lattice`.

### CODE REVIEW

File: gdsfactory/serialization.py [hunk 1]
```diff
@@ -24,13 +24,11 @@
 
 def convert_tuples_to_lists(
     data: dict[str, Any] | list[Any] | tuple[Any, ...] | Any,
-) -> dict[str, Any] | list[Any]:
+) -> dict[str, Any] | list[Any] | Any:
     if isinstance(data, dict):
         return {key: convert_tuples_to_lists(value) for key, value in data.items()}
-    elif isinstance(data, list):
+    elif isinstance(data, list | tuple):
```
Comment (MatthewMckee4) [inline gdsfactory/serialization.py:30]: Note this, feels like if a tuple contains a tuple, they tuple elements should be converted to list too

File: Makefile [hunk 2]
```diff
@@ -38,7 +38,7 @@ test-force:
 	pytest -n auto --force-regen
 
 uv-test: test-data-gds
-	uv run pytest -s -n auto
+	uv run pytest -s -n logical
```
Comment (MatthewMckee4) [inline Makefile:41]: This makes it possible for more cores to be used sometimes, it falls back to auto if it cant detect any logical CPU cores

Comment (joamatab) [reply Makefile:41]: perfect, Thank you Matthew!

**Review by joamatab — APPROVED** [review]
looks great!, thank you Matthew!

**Pull request merged**